modules/encrypt/classes/Kohana/Encrypt/Engine/Mcrypt.php

```diff
@@ -154,11 +64,8 @@ public function __construct($key, $mode, $cipher)
 	 * @param   string  $data   data to be encrypted
 	 * @return  string
 	 */
-	public function encode($data)
+	public function encrypt($data, $iv)
```

Comment:
We should always assume that a programmer programming is a monkey, with no knowledge of encryption. I would therefore suggest not including the IV in encrypt() function, but instead automatically generate the most random IV.